### PR TITLE
Backend Structure Update, now adhering to eslint guidelines (loads of backend testing)

### PR DESCRIPTION
**Pull request by Michiocre**
The main Application function has now been made into a class. The reason is that you are able to create a new Instance of the Application for backend testing.

I also added exceptions to the no-console eslint rule so basic logging is not displayed as a warning during testing. Only permanant logs will be flagged this way so temporary logs still show up as warnings.

### CODE REVIEW

**Comment by Michiocre** [comment]
Also added new Backend-Testing and made it easier to expand on these.

**Comment by Michiocre** [comment]
Suggestion: configure hound for ES6

**Comment by Gabsii** [comment]
@Michiocre please fix all Hound tasks before I review

File: backend/__tests__/fullSuiteTest.test.js [hunk 1]
```diff
@@ -1,70 +1,379 @@
...
+                    });
+                });
+                socket.on('twoRooms', (data) => {
+                    expect(data.oldRoom).toBe(room.id);
+                    socket.emit('twoRooms', {
+                        roomId: room.id,
+                        value: true
+                    });
+                });
+                socket.on('initData', (update) => {
```
[the remark this answers is not in the document]
Comment (Gabsii) [reply backend/__tests__/fullSuiteTest.test.js]: @Michiocre

File: backend/__tests__/fullSuiteTest.test.js [hunk 2]
```diff
@@ -1,70 +1,379 @@
...
+                    
+                    expect(update).toBeDefined();
+                    expect(update.host.name).toBe('jest');
+                    done();
+                });
+                socket.on('errorEvent', (data) => {
+                    expect(data.message).toBe('NO ERROR');
+                    done();
+                });
+                socket.on('nameEvent', (data) => {
```
[the remark this answers is not in the document]
Comment (Gabsii) [reply backend/__tests__/fullSuiteTest.test.js]: @Michiocre

**Comment by Gabsii** [comment]
@Michiocre is this PR ready to merge?

**Comment by Michiocre** [comment]
Im having trouble with the async testing it does work on my pc but the automated build of circle ci fails, i will have to change the tests yet again

File: backend/Classes/App.js [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+*/
+method.theUpdateFunction = async function(socket) {
+    let room = lib.getRoomById(socket.roomId, this.rooms);
+
+    socket.updateCounter.amount += 1;
+
+    if (room !== null) {
+        await room.update(socket.isHost);
+
+        if (socket.updateCounter.amount % 300 === 0 && socket.isHost === true) {
```
Comment (Gabsii) [inline backend/Classes/App.js]: replace numbers with globals constants

File: backend/Classes/Room.js [hunk 4]
```diff
@@ -421,13 +437,8 @@ method.updatePlaylists = async function() {
...
 * @return {boolean} True if completed
 */
-method.getRandomTracks = async function(playlistId, activeTrack) {
-    let playlist = this.getPlaylistById(playlistId);
-    //Load tracks into Playlist if its empty
-    if (!Array.isArray(playlist.tracks)) {
-        playlist.tracks = await this.user.fetchPlaylistTracks(playlist);
-    }
-
+method.getRandomTracks = function(playlist, activeTrack) {
```
Comment (Gabsii) [inline backend/Classes/Room.js]: why isnt this async anymore?

File: backend/__tests__/fullSuiteTest.test.js [hunk 5]
```diff
@@ -1,70 +1,739 @@
...
+
+let ioBack = socketIo(Application.server);
+
+let hostSocket;
+let clientSocket;
+let address = 'http://' + env.ipAddress + ':' + env.backendPort;
+
+ioBack.on('connection', (s) => Application.socketCall(s));
+
+describe('Full Backend Test', () => {
```
Comment (Gabsii) [inline backend/__tests__/fullSuiteTest.test.js]: rename: Server starts properly?

File: backend/Classes/App.js [hunk 6]
```diff
@@ -0,0 +1,592 @@
...
+
+                socket.emit('initData', update);
+            }
+        } else {
+            socket.emit('errorEvent', {message: 'Room was closed'});
+        }
+
+    });
+
+    /**
```
[the remark this answers is not in the document]
Comment (Michiocre) [reply backend/Classes/App.js]: This indentation is not actually off its all part of the socketCall function

File: backend/Classes/Room.js (hunk 4 above)
Comment (Michiocre) [reply backend/Classes/Room.js]: I changed this function so it does not have to make any api calls anymore, the earlier version would (in some cases) make a call to get the tracks of a playlist, i moved the api call somewhere else

File: backend/__tests__/fullSuiteTest.test.js (hunk 5 above)
Comment (Michiocre) [reply backend/__tests__/fullSuiteTest.test.js]: This is the container for the main Backend tests, these are the ones that need a server running for them to work, the tests is not only for the server starting

File: backend/testHelper/spotifyServer.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
+const express = require('express');
```
Comment (Gabsii) [inline backend/testHelper/spotifyServer.js]: maybe change the folder structure from `backend/testHelper/`... to `backend/test/helper` for a more readable code.

File: frontend/config/modules.js [hunk 8]
```diff
@@ -39,7 +39,7 @@ function getAdditionalModulePaths(options = {}) {
     // Otherwise, throw an error.
     throw new Error(
         chalk.red.bold(
-            "Your project's `baseUrl` can only be set to `src` or `node_modules`." +
+            'Your project\'s `baseUrl` can only be set to `src` or `node_modules`.' +
```
Comment (Gabsii) [inline frontend/config/modules.js]: use double quotes if possible

File: frontend/config/webpack.config.js [hunk 9]
```diff
@@ -23,8 +23,8 @@ const getClientEnvironment = require('./env');
 const ModuleNotFoundPlugin = require('react-dev-utils/ModuleNotFoundPlugin');
 const ForkTsCheckerWebpackPlugin = require('react-dev-utils/ForkTsCheckerWebpackPlugin');
 const typescriptFormatter = require('react-dev-utils/typescriptFormatter');
-const eslint = require('eslint');
-const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+//const eslint = require('eslint');
```
Comment (Gabsii) [inline frontend/config/webpack.config.js]: don't comment out eslint

File: frontend/scripts/start.js [hunk 10]
```diff
@@ -53,7 +54,7 @@ if (process.env.HOST) {
         )
     );
     console.log(
-        `If this was unintentional, check that you haven't mistakenly set it in your shell.`
+        'If this was unintentional, check that you haven\'t mistakenly set it in your shell.'
```
Comment (Gabsii) [inline frontend/scripts/start.js]: use double quotes if possible

File: frontend/config/webpack.config.js (hunk 9 above)
Comment (Michiocre) [reply frontend/config/webpack.config.js]: I commented it out because it was not used

**Pull request merged**